model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -499,6 +517,86 @@ async def create_vllm_bundle(
             )
         ).model_bundle_id
 
+    async def create_lightllm_bundle(
+        self,
+        user: User,
+        model_name: str,
+        framework_image_tag: str,
+        endpoint_unique_name: str,
+        num_shards: int,
+        checkpoint_path: Optional[str],
+    ):
+        command = []
+
+        # TODO: incorporate auto calculate max_total_token_num from https://github.com/ModelTC/lightllm/pull/81
+        max_total_token_num = 6000  # LightLLM default
+        if num_shards == 1:
+            max_total_token_num = 15000  # Default for Llama 2 7B on 1 x A10
+        elif num_shards == 2:
+            max_total_token_num = 21000  # Default for Llama 2 13B on 2 x A10
```

Comment:
just curious, why does the total number of tokens increase for 2 shards and then decrease for 3? is there any explanation for this (I expected it to be a linear trend)

Reply:
`max_total_token_num` is the full size of KV cache

Reply:
> why does the total number of tokens increase as shards go up

because we have more memory with more shards, also this is 100% a hack before automatic calculating `max_total_token_num`